pkg/utils/node/predicates.go

```diff
@@ -31,7 +31,28 @@ func IsReadyAndSchedulable(node v1.Node) bool {
 	return false
 }
 
-func IsPastEmptyTTL(node *v1.Node) bool {
+func IsReady(node v1.Node) bool {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady {
+			return condition.Status == v1.ConditionTrue
+		}
+	}
+	return false
+}
+
+func FailsToBecomeReady(node v1.Node) bool {
+	if time.Since(node.GetCreationTimestamp().Time) < (15 * time.Minute) {
+		return false
+	}
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady {
+			return condition.Status == v1.ConditionUnknown && condition.LastTransitionTime.IsZero() && condition.LastHeartbeatTime.IsZero()
```

Comment:
I think checking ConditionUnknown should be sufficient.

Reply:
If it goes from Unknown -> False -> True -> Unknown for some reason ([example[(https://kubernetes.io/docs/concepts/architecture/nodes/#node-controller)), then these times would be nonZero. If we check that they're zero, then we know it has never advanced from Unknown to False.

Reply:
How about we just check heartbeat? If we've never heard a heartbeat, then we know it was never alive.

Reply:
I wonder if it's possible for that to get erased. Can we ensure that status will be never be nulled? What if there's some issue with data retention?